compiler/AST/AggregateType.cpp

```diff
@@ -2159,15 +2165,10 @@ bool AggregateType::needsConstructor() const {
   }
 
   AggregateType* thisNC = const_cast<AggregateType*>(this);
-  ModuleSymbol* mod = thisNC->getModule();
 
-  // For now, always generate a default constructor for types in the internal
-  // and library modules
-  if (mod && (mod->modTag == MOD_INTERNAL || mod->modTag == MOD_STANDARD))
-    return true;
-  else if (initializerStyle == DEFINES_CONSTRUCTOR)
+  if (initializerStyle == DEFINES_CONSTRUCTOR)
     return true;
-  else if (fUserDefaultInitializers)
+  else if (fUserDefaultInitializers && this != dtObject)
```

Comment:
I think ideally dtObject would be using initializers as well.  What is the issue that is preventing this?

Reply:
I think dtObject still needs to have a constructor so that user classes with constructors can compile. When constructors are fully deprecated, dtObject will have an initializer.

Reply:
Oh right, good point!